### DISCUSSION TITLE
Auto-updation of app on linux and macOS

### DISCUSSION THREAD

**Discussion by Amogh-Gurudatta** [ThioJoe/YT-Spammer-Purge#315] [Ideas]
It's hard to keep downloading the files again and again as you are updating the app constantly. Could you please add the feature of auto-updating for Linux and macOS users?

**Comment by rbeer** [comment]
Just use git?

`git clone https://github.com/ThioJoe/YT-Spammer-Purge` creates a `YT-Spammer-Purge` dir with the program in it and whenever you want to update it, you just run `git pull` in that directory.

**Reply by Amogh-Gurudatta** [reply]
hmm yes, that'll work. thanks.

**Reply by Amogh-Gurudatta** [reply]
wait, but how is this possible? I used @rbeer's method.
![screenshot](https://user-images.githubusercontent.com/77231061/148024924-3572ec29-1c25-4eca-b45c-4fc457952e6d.png)
-

**Reply by rbeer** [reply]
This is possible, because when using git pulls, you are always at the very latest version of the code as it lives in this repository.
The update via program looks at releases.

Thio has pushed one more change, after he released 2.6.1. Compare these list of commits:

https://github.com/ThioJoe/YT-Spammer-Purge/commits/v2.6.1
https://github.com/ThioJoe/YT-Spammer-Purge/commits/main

7574baf is on master, after 2.6.1 was released.

There ~has to be some~ is a string that defines the version which he changed to 2.6.2, but didn't do a release.
https://github.com/ThioJoe/YT-Spammer-Purge/blob/7574baf6effd33568e7113f713fc2e34b693fd06/YouTubeSpammerPurge.py#L39

If  you want to stick to releases, you can always pull, look at the list of tags and then checkout the state at the latest release tag:
```bash
git pull
git tags -l
> ...
v2.4.0
v2.4.1
v2.4.2
v2.5.0
v2.5.1
v2.5.2
v2.5.3
v2.5.4
v2.5.5
v2.6.0
v2.6.1
git checkout v2.6.1
> Note: switching to 'v2.6.1'.
...
```

Be aware though that you will have to switch back to the `main` branch, before you do the next `pull`
```bash
git checkout main
git pull
[and then the rest from above for the newest version]
```
